### PR TITLE
Split Right model into Copyright and License

### PR DESCRIPTION
**Pull request by matt-bernhardt**
Finally, this is ready for code review. I think. I have kept the changes in three commits:

1. Adds the Copyright model and associated changes
2. Adds the License model and associated changes
3. Removes the Right model and associated changes.

Looking at the diff's individually and together, I'm not sure if it is easier to review piece-by-piece, or all together. Reviewer's choice, I guess.

Some specific questions:

* There is currently no guidance text below the license field on the "Submit your thesis" form. I'm not sure if there needs to be?
* Some changes happened to the tests because we're going from a required Right field to optional fields for Copyright and License. I hope those make sense and are justified - but I'd love feedback on this.

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

YES

#### Includes new or updated dependencies?

NO

### CODE REVIEW

**Review by JPrevost — COMMENTED** [review]
Please take a look at the seeds feedback as I think it'll be better to adjust now to prevent us from ourselves in the future if we accidentally re-run seeds after changes have been made.

If we choose not to take a stance on what to do about deleting of licenses and copyright in this ticket (probably sane) we should just make sure we have an backlog ticket with a brief summary of the problem.

The remaining comment is primarily informational.

File: app/dashboards/license_dashboard.rb
Comment (matt-bernhardt) [reply app/dashboards/license_dashboard.rb:69]: Ok, that makes sense - will change.

File: db/seeds.rb
Comment (matt-bernhardt) [reply db/seeds.rb:7]: Ooh, yeah... I'll re-create locally and try that approach.

File: test/integration/thesis_test.rb
Comment (matt-bernhardt) [reply test/integration/thesis_test.rb:3]: I'll try writing a test for this scenario and see what happens? Spawning follow-up tickets could be the right step, or if this is simple to adjust I can just do it now.

**Review by JPrevost — APPROVED** [review]
We've opened a spike to get decisions around desired behavior around deleting / editing Copyright and License records.

https://mitlibraries.atlassian.net/browse/ETD-198

👍🏻

**Comment by matt-bernhardt** [comment]
I've tagged Helen for review against the data model. Will merge once she approves.

**Pull request merged**
